src/engine/GroupBy.cpp

```diff
@@ -1160,14 +1339,23 @@ void GroupBy::evaluateAlias(
 
     // Substitute in the results of all aggregates contained in the
     // expression of the current alias, if `info` is non-empty.
-    substituteAllAggregates(info, evaluationContext._beginIndex,
-                            evaluationContext._endIndex, aggregationData,
-                            result, localVocab);
+    std::vector<std::unique_ptr<sparqlExpression::SparqlExpression>>
+        originalChildren = substituteAllAggregates(
+            info, evaluationContext._beginIndex, evaluationContext._endIndex,
+            aggregationData, result, localVocab);
 
     // Evaluate top-level alias expression
     sparqlExpression::ExpressionResult expressionResult =
         alias.expr_.getPimpl()->evaluate(&evaluationContext);
 
+    // Restore original children
+    for (size_t i = 0; i < info.size(); ++i) {
+      auto& aggregate = info.at(i);
+      auto parentAndIndex = aggregate.parentAndIndex_.value();
+      parentAndIndex.parent_->replaceChild(parentAndIndex.nThChild_,
+                                           std::move(originalChildren.at(i)));
+    }
```

Comment:
Review question @joka921 should we add a template parameter (or a bool flag) that avoids restoring the original children for the hash map optimization code, where it is not required at all.